File: Mobiles/ios/iOS-MobileO11yDemo/Features/CarConnectionAction/Models/Car.swift
//
//  Car.swift
//  iOS-MobileO11yDemo
//
//  Created by Robert Magnusson on 04.06.24.
//

import Foundation

struct Car {
    let info: CarInfo
    let carDoorStatus: CarDoorStatus
}

class CarFactory {
    func getRandomCar() -> Car {
        return Car(
            info: CarInfoFactory().getRandomCarInfo(),
            carDoorStatus: CarDoorStatus(status: "locked")
        )
    }
}

File: Mobiles/flutter/lib/core/car/domain_layer/car.dart
import 'package:flutter_mobile_o11y_demo/core/car/domain_layer/car_door_status.dart';
import 'package:flutter_mobile_o11y_demo/core/car/domain_layer/car_info.dart';

class Car {
  const Car({
    required this.info,
    required this.doorStatus,
  });

  final CarInfo info;
  final CarDoorStatus doorStatus;
}

class CarFactory {
  CarFactory({
    required this.carInfoFactory,
  });

  final CarInfoFactory carInfoFactory;

  Car getRandomCar() {
    return Car(
      info: carInfoFactory.getRandomCarInfo(),
      doorStatus: const CarDoorStatus(isLocked: false),
    );
  }
}
